core/execution/liquidation/snapshot.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2023 Gobalsky Labs Limited
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package liquidation
+
+import (
+	"context"
+
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/proto"
+)
+
+func (e *Engine) Namespace() types.SnapshotNamespace {
+	return types.LiquidationSnapshot
+}
+
+func (e *Engine) Keys() []string {
+	return []string{e.mID}
+}
+
+// GetState must be thread-safe as it may be called from multiple goroutines concurrently!
+func (e *Engine) GetState(key string) ([]byte, []types.StateProvider, error) {
+	if key != e.mID {
+		return nil, nil, types.ErrSnapshotKeyDoesNotExist
+	}
+	if e.stopped {
+		return nil, nil, nil
+	}
+	payload := e.buildPayload()
+
+	s, err := proto.Marshal(payload.IntoProto())
+	return s, nil, err
+}
+
+func (e *Engine) LoadState(ctx context.Context, pl *types.Payload) ([]types.StateProvider, error) {
+	if e.Namespace() != pl.Namespace() {
+		return nil, types.ErrInvalidSnapshotNamespace
+	}
+
+	switch d := pl.Data.(type) {
+	case *types.LiquidationNode:
+		e.mID = d.MarketID
+		e.pos.open = d.NetworkPos
+		e.nextStep = d.NextStep
+		if d.Config != nil {
+			e.cfg = d.Config.DeepClone()
+		} else {
+			// @NOTE this can be removed after protocol upgrade has completed
```

Comment:
Minor: I don't think we'd every hit this because they'll be no payload in the protocol-upgrade snapshot for us to load from. I think we're cover by the bit in `NewMarketFromSnapshot()` though, so thats good.

Reply:
yes, the places where nil values are being replaced with the legacy strat is overkill ATM. Part of the upcoming PR is requiring the liquidation strategy to be set in the proposals, and then I'm planning to go through all these bits and bobs, and remove dead code.